Microsoft.NET.Build.Containers/ParseContainerProperties.cs

```diff
@@ -57,16 +58,51 @@ public ParseContainerProperties()
         ParsedContainerTag = "";
         NewContainerRegistry = "";
         NewContainerImageName = "";
-        NewContainerTag = "";
+        NewContainerTags = Array.Empty<string>();
+    }
+
+    private static bool ParsePropertyAsArray(string input, [NotNullWhen(true)] out string[]? items) {
+        if(String.IsNullOrEmpty(input)) {
+            items = null;
+            return false;
+        } else {
+            items = input.Split(';');
+            return true;
+        }
+    }
+
+    private static bool TryValidateTags(string[] inputTags, out string[] validTags, out string[] invalidTags) {
+        var v = new List<string>();
+        var i = new List<string>();
+        foreach (var tag in inputTags) {
+            if (ContainerHelpers.IsValidImageTag(tag)) {
+                v.Add(tag);
+            } else {
+                i.Add(tag);
+            }
+        }
+        validTags = v.ToArray();
+        invalidTags = i.ToArray();
+        return invalidTags.Count() == 0;
     }
 
     public override bool Execute()
     {
 
-        if (!string.IsNullOrEmpty(ContainerImageTag) && !ContainerHelpers.IsValidImageTag(ContainerImageTag))
+        string[] validTags;
+        if (!string.IsNullOrEmpty(ContainerImageTag) && ParsePropertyAsArray(ContainerImageTag, out var inputTags) && !TryValidateTags(inputTags, out var valids, out var invalids))
         {
-            Log.LogError($"Invalid {nameof(ContainerImageTag)}: {0}", ContainerImageTag);
-            return !Log.HasLoggedErrors;
+            if (invalids.Any()) {
+                if (invalids.Length == 1) {
+                    Log.LogError(null, "CONTAINER003", "Container.InvalidTag", null, 0, 0, 0, 0, $"Invalid {nameof(ContainerImageTag)} provided: {0}. {nameof(ContainerImageTag)} must be a semicolon-delimited list of valid image tags. Image tags must be alphanumeric, underscore, hyphen, or period.", invalids[0]);
+                } else {
+                    Log.LogError(null, "CONTAINER003", "Container.InvalidTag", null, 0, 0, 0, 0, $"Invalid {nameof(ContainerImageTag)}s provided: {0}. {nameof(ContainerImageTag)} must be a semicolon-delimited list of valid image tags. Image tags must be alphanumeric, underscore, hyphen, or period.", String.Join(", ", invalids));
+                }
+                return !Log.HasLoggedErrors;
+            }
+            validTags = invalids;
```

Comment:
```suggestion
            validTags = valids;
```

Reply:
well that's embarassing! I think overall I might want to refactor this - instead of using nullability hijinks, returning a tuple of valid-and-invalid tags from the parse might be good enough.